src/main/kotlin/kotliquery/Session.kt

```diff
@@ -19,7 +27,7 @@ open class Session(
     open val autoGeneratedKeys: List<String> = listOf(),
     var transactional: Boolean = false,
     open val strict: Boolean = false
-) : AutoCloseable {
+) : Closeable {
```

Comment:
This is fine but can you share the reason why I've changed this interface to sub interface?

Reply:
Allows to utilize `use` function build-in Kotlin

https://stackoverflow.com/questions/26969800/try-with-resources-in-kotlin 

If you take a look in its source code it is way reacher than `LoanPattern`